Materialized View feature

kyungmax opened the pull request:
I have implemented simple features for Materialized View (No query rewrite and incremental refresh)
1. `CREATE MATERIALIZED VIEW`
2. `REFRESH MATERIALIZED VIEW`
3. `DROP MATERIALIZED VIEW`

Implementation
- Mainly follows the CTAS flow
- Declare a new catalog entry `MatViewCatalogEntry` with type `MATVIEW_ENTRY` and additional field `query` for later REFRESH MATERIALIZED VIEW. But it also basically follows how DuckDB treats TABLE_ENTRY

Please review and let me know if the direction and implementation make sense.

kyungmax committed:
implementation of MView feature in DuckDB (Works mainly from @xnge)

kyungmax committed:
feat: Implement to support REFRESH MATERIALIZED VIEW

kyungmax committed:
trying to implement bind

kyungmax committed:
trying to implement bind, planning

kyungmax committed:
I think creating a mview can be delegated to create table

kyungmax committed:
create mview WAL handle

kyungmax committed:
Allow MATVIEW_ENTRY

kyungmax committed:
store select query for later refresh

kyungmax committed:
allow MATVIEW_ENTRY

kyungmax committed:
MatViewCatalogEntry for matview

kyungmax committed:
Merge branch 'main' into feature/mview

# Conflicts:
#	src/planner/table_binding.cpp
#	third_party/libpg_query/src_backend_parser_gram.cpp

kyungmax committed:
retain select query of CREATE MATERIALIZED VIEW for later REFRESH MATERIALIZED VIEW

kyungmax committed:
use CTAS flow for create matview

kyungmax committed:
implement REFRESH MATERIALIZED VIEW

kyungmax committed:
prevent delete from MATVIEW

kyungmax committed:
misc modification

kyungmax committed:
Merge branch 'main' into feature/mview

kyungmax committed:
clean up unnecessary parts

kyungmax committed:
change the matview entry name

kyungmax committed:
omitted '&' when returning to the original implementation

Maxxen commented on src/include/duckdb/common/enums/catalog_type.hpp:
```diff
@@ -18,14 +18,15 @@ namespace duckdb {
-	COLLATION_ENTRY = 7,
-	TYPE_ENTRY = 8,
-	DATABASE_ENTRY = 9,
+	MATVIEW_ENTRY = 2,
```
I don't think you can change the orders of these without breaking serialization and backwards/compatability.

kyungmax replied:
@Maxxen Thank you for the review. I'll append the new entry to the last.

Maxxen commented on src/catalog/catalog_entry/CMakeLists.txt:
```diff
@@ -17,7 +17,9 @@ add_library_unity(
   table_function_catalog_entry.cpp
-  view_catalog_entry.cpp)
+  view_catalog_entry.cpp
+  matview_catalog_entry.cpp
```
Im not sure I love "MatView", we generally try to avoid abbreviations in class names - but I guess its not important.

kyungmax replied:
Well 'MateraizliedView..' for 'MatView..' is quite long but I also agree to be clear as possible. I'll change them all.

Tishj replied:
I wasn't going to comment, but I have the same thought.
We're still paying the mental price of DelimJoin, being short for Duplicate Eliminated Join, as opposed to meaning Deliminator Join

Maxxen commented:
I haven't tested anything yet but left some comments. But cool work!

kyungmax committed:
Apply Review Feedback:
- Change 'MatView' abbreviation to 'MaterializedView' for clear meaning
- Fix the order of catalog entry type for compatibility
- Apply format fix

kyungmax committed:
Merge branch 'main' into feature/mview

kyungmax committed:
fix: unittest fail (Copy operation)

kyungmax committed:
fix: CI fail
